src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,39 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Machine;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        Map<String, MachineProducer<? extends Machine>> producers = new HashMap<>();
+        producers.put("Bulldozer", new BulldozerProducer());
+        producers.put("Excavator", new ExcavatorProducer());
+        producers.put("Truck", new TruckProducer());
+        String typeKey = type.getSimpleName();
+        if (producers.containsKey(typeKey)) {
+            return new ArrayList<>(producers.get(typeKey).get());
```

Comment:
Your producer returns list instance
```suggestion
            return producers.get(typeKey).get();
```